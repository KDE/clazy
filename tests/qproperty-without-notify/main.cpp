#include <QtCore/QObject>

class MyObj : public QObject
{
    Q_OBJECT
    Q_PROPERTY(int foo READ foo) // Warn
    Q_PROPERTY(int foo1 READ foo1 CONSTANT) // OK
    Q_PROPERTY(int foo2 READ foo2 NOTIFY fooChanged) // KO
    Q_PROPERTY(int foo3 READ foo3 WRITE setFoo) // Warn
};

class MyGadget // No signals allowed
{
    Q_GADGET
    Q_PROPERTY(int foo READ foo) // Ok
    Q_PROPERTY(int foo2 READ foo2 WRITE setFoo2)
    Q_PROPERTY(int foo3 MEMBER foo3)
};

class MyObj2 : public QObject
{
    Q_OBJECT
    Q_PROPERTY(int foo READ foo) // Warn
    Q_PROPERTY(int foo1 READ foo1) // Warn
    Q_PROPERTY(int foo2 READ foo2
        NOTIFY foo2Changed) // Multiline case. OK
    Q_PROPERTY(int foo3 READ foo3 NOTIFY
        foo3Changed) // Another multiline case. OK
};
