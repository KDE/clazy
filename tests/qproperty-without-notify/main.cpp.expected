qproperty-without-notify/main.cpp:6:5: warning: Q_PROPERTY should have either NOTIFY or CONSTANT [-Wclazy-qproperty-without-notify]
qproperty-without-notify/main.cpp:9:5: warning: Q_PROPERTY should have either NOTIFY or CONSTANT [-Wclazy-qproperty-without-notify]
qproperty-without-notify/main.cpp:23:5: warning: Q_PROPERTY should have either NOTIFY or CONSTANT [-Wclazy-qproperty-without-notify]
qproperty-without-notify/main.cpp:24:5: warning: Q_PROPERTY should have either NOTIFY or CONSTANT [-Wclazy-qproperty-without-notify]
